bot/services/tag_service.py

```diff
@@ -23,22 +25,55 @@ async def on_message_recieved(self, message: discord.Message) -> None:
         pattern = f'(^| )[{TAG_PREFIX}](?P<name>\\w+)($| )'
         found_name = re.search(pattern, message.content)
         
+        validTagFound = False
         if not found_name:
             return
 
-        name = found_name.groupdict()['name'].lower()
+        tagsContent = ''
         
-        if not await repo.check_tag_exists(name, message.guild.id):
+        for match in re.finditer('(^| ?)[$](\w+)($| )', message.content, re.S):
+            end = match.end()
+            
+            name = found_name.groupdict()['name'].lower()
+            if not await repo.check_tag_exists(name, message.guild.id):
+                found_name = re.search(pattern, message.content[end: ])
+                continue
+            validTagFound = True
+            tagsContent += await repo.get_tag_content(name, message.guild.id) + '\n'
+            await repo.increment_tag_use_counter(name, message.guild.id)
+            log.info(f'Tag "{found_name}" invoked in guild: {message.guild.id} by: {message.author.id}')
+                    
+            found_name = re.search(pattern, message.content[end: ])
+            if(found_name):
+                tagsContent += '-----\n'
+
+        if not validTagFound:
             return
+            
+        #If length of all tags is greater than the threshold, sends it to the paginate service, otherwise sends as a normal message
+        if len(tagsContent) > TAG_PAGINATE_THRESHOLD:
```

Comment:
instead of this you cna use this helper function to chunk your string https://github.com/ClemsonCPSC-Discord/ClemBot/blob/7f5420cbd078b0727ae718b05546c24fda8c08aa/bot/cogs/source_code_cog.py#L160

Reply:
Implemented